packages/documentation/copy/en/reference/JSX.md

```diff
@@ -386,11 +444,30 @@ You can customize the type by specifying the `JSX.Element` interface.
 However, it is not possible to retrieve type information about the element, attributes or children of the JSX from this interface.
 It is a black box.
 
+## The JSX function return type
+
+By default, JSX functions must return the `JSX.Element` type. However, this doesn’t always represent runtime behaviour. As of TypeScript 5.1, you can specify `JSX.ElementType` to override what is a valid JSX component type. The default looks something like this.
```

Comment:
We should mention here that the type of the props is not extracted from this type at the moment. Props type inference still only looks at the first argument of functions or constructors, otherwise looks up props from `JSX.IntrinsicElements`